modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManager.java

```diff
@@ -39,18 +34,15 @@ public interface SqlSchemaManager {
      * Returns a table by given id.
      *
      * @param id An id of required table.
-     *
      * @return The table.
      */
     IgniteTable tableById(int id);
 
     /**
-     * Wait for {@code ver} schema version, just a stub, need to be removed after IGNITE-18733.
-     */
-    CompletableFuture<?> actualSchemaAsync(long ver);
-
-    /**
-     * Returns a required schema if specified, or default schema otherwise.
+     * Returns a future to wait for given SQL schema version readiness.
+     *
+     * @param version SQL schema version to wait.
+     * @return Operation future.
      */
-    SchemaPlus activeSchema(@Nullable String name, long timestamp);
+    CompletableFuture<?> schemaReadyFuture(long version);
```

Comment:
I am confused by the type of the return value of the future, I propose to change it to `Void`, we will wait for the version of the scheme and then we will receive it separately as we need.